### ISSUE TITLE
No viable error handler for non-deferred interactions 

### ISSUE THREAD

**Issue by SGA-A** [SGA-A/c2c#4]
## Summary
For commands that contain an interaction that is not being deferred, and an error is being handled, the interaction is successfully deferred to respond accordingly but it raises the error:

```discord.errors.InteractionResponded: This interaction has already been responded to before```

## Additional Context
The error handler for interaction-based contexts automatically tries to defer the response if it is not already. This is done by acknowledging if the interaction is done or not:

https://github.com/SGA-A/c2c/blob/131f55acfd526238310fcfce8a52cdfa04b4f00e/cogs/slash-events.py#L23-L24

We are looking into the issue, but note that this does not actually prevent the command from being executed after an error was found in the error handler.

**Comment by SGA-A** [comment]
We are still encountering this issue in rare situations. 
We are looking into this. 
No ETA can be given.

**Comment by SGA-A** [comment]
The causes have been identified. 
The issue was occuring due to responding to an interaction despite already sending a response prior. 
This is not news to most of us however, as we are aware of this.
Deferring an interaction was one way this could happen, which was a method I was not aware of myself until I delved into the docs.
Seeing as the main root cause has been identified, I'm closing this issue, the title of the issue has been resolved and there are no additional changes needed.
